x509/cert_validations.go

```diff
@@ -0,0 +1,166 @@
+package x509
+
+import (
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"fmt"
+)
+
+// VerifyCACertificate validates root/intermediate certificates according to this spec:
+// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#root-and-intermediate-ca-certificates
+func VerifyCACertificate(cert *x509.Certificate, expectedPathLen int) error {
+	if err := verifyCABasicConstraints(cert, expectedPathLen); err != nil {
+		return err
+	}
+	return verifyCAKeyUsage(cert)
+}
+
+// VerifyCodeSigningCertificate validates code-signing certificates according to this spec:
+// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#leaf-certificates
+func VerifyCodeSigningCertificate(cert *x509.Certificate) error {
+	return verifyCertificate(cert, x509.ExtKeyUsageCodeSigning)
+}
+
+// VerifyTimeStampingCertificate validates time-stamping certificates according to this spec:
+// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#leaf-certificates
+func VerifyTimeStampingCertificate(cert *x509.Certificate) error {
+	return verifyCertificate(cert, x509.ExtKeyUsageTimeStamping)
+}
+
```

Comment:
1. Can we please update [`ValidateCertChain`](https://github.com/notaryproject/notation-core-go/blob/main/x509/cert.go#L49) method with this validations?
2. If we do 1st then we dont need to make these methods public.
3. Should we move `ValidateCertChain` from `certs.go` to `cert_validations.go`